### ISSUE TITLE
String Representation of "x*(y<1)" is Wrong

### ISSUE THREAD

**Issue by rdbyk** [symengine/symengine#1450]
The parsing (`basic_parse`) seems to be Ok, but the string representation (`basic_str`) has an issue, cf. below
```
--> Sym("x*y<1")
 ans  =

   x*y < 1
                                                                                                                      (1:StrictLessThan)

--> GetArgs(ans)
 ans  =

   < x*y, 1 >
                                                                                                                               (1:<...>)

--> Sym("x*(y<1)")
 ans  =

   x*y < 1 // Not OK, should read "x*(y < 1)"
                                                                                                                                 (1:Mul)

--> GetArgs(ans)
 ans  =

   < x, y < 1 >
                                                                                                                               (1:<...>)
```

**Comment by isuruf** [comment]
Thanks for the bug report. All the `Relational` operators needs to have `PrecdenceEnum::Add` like https://github.com/symengine/symengine/blob/master/symengine/printer.h#L20

**Comment by rdbyk** [comment]
Thanks, I have tried that, but it seems that `Relational` operators need their own precedence level less than `PrecdenceEnum::Add` ?
PrecdenceEnum::Add

**Comment by isuruf** [comment]
You are right. It needs something lower.

**Comment by rdbyk** [comment]
Sorry for closing it by accident ..... So do you think one should introduce `PrecdenceEnum::Relational`? I could try to fix it.

--> Sym("x*(y<1)")
 ans  =

   x*(y < 1)
                                                                                                                                  (1:Mul)

--> GetArgs(ans)
 ans  =

   < x, y < 1 >
                                                                                        
                                                                                                                                  (1:Add)

--> GetArgs(ans)
 ans  =

   < y < 1, x >
```

**Comment by isuruf** [comment]
Yes. If you can send a PR with a test, I'll be happy to review

**Comment by isuruf** [comment]
Fixed in #1451
